### DISCUSSION TITLE
Building a project through a GUI Windows app using Autohotkey

### DISCUSSION THREAD

**Discussion by skymen** [community/community#34126] [Actions] [answered]
Hi,

I'm trying to run a build script I have locally but on Github actions.

I have an AutoHotKey compiled script (so it doesn't need AHK to run) that takes in a handful of params, opens a program and interacts with its GUI.
At the moment, this is the only way I found to build properly. That program does not have a CLI version, nor am I able to reverse engineer it at the moment to rewrite the export process.

My script runs fine on my local computer, and on github actions it does not produce any error, however in the release build, the export folder is empty, which should mean that my script failed to run.

As far as I know, I have no way to see what is going on. It would be very useful if I could just connect to the server and look at what's happening, but I can't it seems.

I looked around, and from what I can tell, Windows Server 2022 can come with or without a GUI and I have no idea what github action uses since it's not specified anywhere.

Anyone has any ideas as to what I should try?

**Comment by Simran-B** [comment]
I suppose it's a headless Windows (no GUI) but it's still capable of RDP in principle.

It would be helpful if you could share what program you try to automate.

**Reply by skymen** [reply]
I'm trying to run Construct 2, a Windows Game engine.

With some more work, it turns out that it does have a GUI since I am able to take screenshots. The software was just stuck on a popup that never appeared to me in my local testing

**Comment by skymen** [comment, marked as answer]
Turns out Windows Server comes with a GUI, and turns out my script AHK worked just fine, I just had an obscure edge case to cover that i only discovered by taking screenshots of the GUI as my script was running
